src/features/PhoneAuth/ui/PhoneAuth.tsx

```diff
@@ -0,0 +1,91 @@
+import { Button, Flex, Text } from '@mantine/core';
+import { auth } from 'shared/firebase/firebase';
+import { useCallback, useState } from 'react';
+import { ConfirmationResult, RecaptchaVerifier, signInWithPhoneNumber } from 'firebase/auth';
+import { useUser } from 'entities/User';
+import PhoneInput from 'react-phone-number-input';
+import ReactCodeInput from 'react-code-input';
+
+export const PhoneAuth = () => {
+    const { setUser } = useUser();
+    const [phone, setPhone] = useState<string>('');
+    const [code, setCode] = useState<string>('');
```

Comment:
why don't you use useForm?

Reply:
+, and use redux state instead of useState, use this hook only for simple logic